### PR TITLE
Fix return type in getCustomerIsGuest method.

### LINKED ISSUE #29529: [Issue] Fix return type in getCustomerIsGuest method.
This issue is automatically created based on existing pull request: magento/magento2#28920: Fix return type in getCustomerIsGuest method.

---------
**Description**

The method "**_getCustomerIsGuest_**" returned a value of type string ("0" or "1") making the comparison of this method within the **_CheckCartCheckoutAllowance_** class always return true, because the comparison is Boolean.

**Manual testing**

Send a GraphQl request with the mutations "_setShippingAddressesOnCart_" or "_setPaymentMethodOnCart_", testing with the logged in customer or not, and follow the behavior of the **_CheckCartCheckoutAllowance_** class of the "**execute**" method.

### PR DESCRIPTION
**Pull request by luisdev1**
**Description**

The method "**_getCustomerIsGuest_**" returned a value of type string ("0" or "1") making the comparison of this method within the **_CheckCartCheckoutAllowance_** class always return true, because the comparison is Boolean.

**Manual testing**

Send a GraphQl request with the mutations "_setShippingAddressesOnCart_" or "_setPaymentMethodOnCart_", testing with the logged in customer or not, and follow the behavior of the **_CheckCartCheckoutAllowance_** class of the "**execute**" method.

### Resolved issues:
1. [x] resolves magento/magento2#29529: Fix return type in getCustomerIsGuest method.

### CODE REVIEW

**Comment by luisdev1** [comment]
@magento give me test instance
@magento give me 2.4-develop instance
@magento run all tests

**Comment by magento-engcom-team** [comment]
Hi @luisdev1. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @luisdev1, here is your [new Magento instance](https://pr-28920.instances.magento-community.engineering).
Admin access: [https://pr-28920.instances.magento-community.engineering/admin_5421](https://pr-28920.instances.magento-community.engineering/admin_5421)
Login: `3516da4e` Password: `7dd0942aa3e1`
Instance will be terminated in up to 3 hours.

**Comment by luisdev1** [comment]
@magento WebAPI Tests

**Comment by luisdev1** [comment]
@lbajsarowicz @eduard13 Can you check this correction? Thank you so much friends.

File: app/code/Magento/QuoteGraphQl/Model/Cart/CheckCartCheckoutAllowance.php [hunk 1]
```diff
@@ -39,7 +39,7 @@ public function __construct(
      */
     public function execute(Quote $quote): void
     {
-        if (false === $quote->getCustomerIsGuest()) {
+        if (false === (bool)$quote->getCustomerIsGuest()) {
```
Comment (lbajsarowicz) [inline app/code/Magento/QuoteGraphQl/Model/Cart/CheckCartCheckoutAllowance.php]: Why not easier:
```suggestion
        if (!$quote->getCustomerIsGuest()) {
```

**Review by lbajsarowicz — COMMENTED** [review]
There is an easier way, in my opinion.

**Comment by luisdev1** [comment]
@lbajsarowicz Yes you are right! Thank you very much, I updated the pull request.

**Comment by lbajsarowicz** [comment]
@magento run all tests

Just for formal purposes.

**Review by lbajsarowicz — DISMISSED** [review]
That is a minor change that does not need additional automated tests to be written.

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7884](https://jira.corp.magento.com/browse/ENGCOM-7884) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@luisdev1 thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by luisdev1** [comment]
@lbajsarowicz Thanks for the assistance and follow-up!

**Comment by sidolov** [comment]
@magento create issue

**Review by lenaorobei — CHANGES_REQUESTED** [review]
I think that this approach is not the proper way of handling such issues. As per contract interface method `getCustomerIsGuest` returns `bool`. In case if contract is broken somewhere, we need to identify this place and fix there.

**Comment by lenaorobei** [comment]
Closing this PR due to inactivity. Please feel free to reopen if you are going to continue working on it.
Thank you.

**Comment by lenaorobei** [comment]
Hi @luisdev1. I think we need to to keep strict comparison where it is possible.
Such changes may lead to destructive security violations.

**Comment by lbajsarowicz** [comment]
@lenaorobei This does not resolve the issue reported by OP.
@luisdev1  Could you create an issue that describes the use-cases when you get `"0"` and `"1"` and assign directly to @lenaorobei to address that internally?

**Comment by luisdev1** [comment]
I understand @lenaorobei . The method is found in the Quote class.

Would it be the case that we add a conversion in the return of the method?

File: app/code/Magento/Quote/Model/Quote.php
Line: 705

public function getCustomerIsGuest ()
{
     return (bool) $ this -> _ getData (self :: KEY_CUSTOMER_IS_GUEST);
}

**Pull request closed without merging**